src/main/java/jobmanager/messaging/handler/AbortJobHandler.java

```diff
@@ -45,46 +53,75 @@ public AbortJobHandler(MongoAccessor accessor, PiazzaLogger logger) {
 		this.logger = logger;
 	}
 
-	public void process(ConsumerRecord<String, String> consumerRecord) {
+	public void process(ConsumerRecord<String, String> consumerRecord, String eventId, String workflowUrl) {
 		// Changing the Status in the Job Table to Aborted
 		try {
 			ObjectMapper mapper = new ObjectMapper();
 			PiazzaJobRequest job = mapper.readValue(consumerRecord.value(), PiazzaJobRequest.class);
+
 			// Update the status of the Job. Only update if the Job is currently
 			// pending or running. Otherwise, we cannot cancel a completed Job.
-			String jobId = ((AbortJob)job.jobType).jobId;
+			String jobId = ((AbortJob) job.jobType).jobId;
 			Job jobToCancel = accessor.getJobById(jobId);
 
-			if( jobToCancel.submitterUserName == null ) {
+			if (jobToCancel.submitterUserName == null) {
 				logger.log(String.format("Could not Abort Job %s because it does not have a user associated with it", jobId),
 						PiazzaLogger.WARNING);
 				return;
-				
-			} else if ( !jobToCancel.submitterUserName.equals(job.userName) ){
+
+			} else if (!jobToCancel.submitterUserName.equals(job.userName)) {
 				logger.log(String.format("Could not Abort Job %s because it was not created by user requesting abort.", jobId),
 						PiazzaLogger.WARNING);
 				return;
 			}
-			
+
 			String currentStatus = jobToCancel.status;
-			if ((currentStatus.equals(StatusUpdate.STATUS_RUNNING))
-					|| (currentStatus.equals(StatusUpdate.STATUS_PENDING))
+			if ((currentStatus.equals(StatusUpdate.STATUS_RUNNING)) || (currentStatus.equals(StatusUpdate.STATUS_PENDING))
 					|| (currentStatus.equals(StatusUpdate.STATUS_SUBMITTED))) {
-				accessor.getJobCollection().update(DBQuery.is("jobId", jobId),
-						DBUpdate.set("status", StatusUpdate.STATUS_CANCELLED));
-				logger.log(
-						String.format("Aborted the Job %s of Abort Job ID %s", jobId, consumerRecord.key()),
-						PiazzaLogger.INFO);
+				accessor.getJobCollection().update(DBQuery.is("jobId", jobId), DBUpdate.set("status", StatusUpdate.STATUS_CANCELLED));
+				logger.log(String.format("Aborted the Job %s of Abort Job ID %s", jobId, consumerRecord.key()), PiazzaLogger.INFO);
 			} else {
-				logger.log(String.format("Could not Abort Job %s because it is no longer running.", jobId),
-						PiazzaLogger.WARNING);
+				logger.log(String.format("Could not Abort Job %s because it is no longer running.", jobId), PiazzaLogger.WARNING);
 			}
 
+			// post request to workflow
+			dispatchWorkflowEvent(jobToCancel, eventId, workflowUrl, "Job Cancelled");
```

Comment:
This must be wrapped individually in a try-catch, so as now to affect the flow of the normal function handling.